src/data/storage.js

```diff
@@ -1,12 +1,41 @@
-const clearFns = []
+const customClearFns = []
 
 export function onClear(fn) {
-  clearFns.push(fn)
+  customClearFns.push(fn)
 }
 
-export function clear() {
-  while (clearFns.length) {
-    clearFns.pop()()
+export function runCustomClears() {
+  while (customClearFns.length) {
+    customClearFns.pop()()
   }
 }
 
+const SHORT = sessionStorage
+const LONG = localStorage
+export const LIFESPAN = Object.freeze({
+  SHORT,
+  LONG,
+})
+
+export function save(key, value, LIFESPAN) {
```

Comment:
Doesn't support same-named keys in both places. Perhaps throw if they try to set the same key in both places.

Reply:
https://github.com/silinternational/wecarry-ui/pull/129/commits/d1027e022ec69a07170ccbcbff80fbaf6b0bb3ce